pkg/types/intstr/intstr.go

```diff
@@ -0,0 +1,70 @@
+// Package intstr implements custom types
+package intstr
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// ValueType defines the type of a IntOrString value
+type ValueType int
+
+const (
+	// IntValue is a IntOrString that represents an integer value
+	IntValue ValueType = iota
+	// StringValue is a IntOrString 	that represents a string value
+	StringValue
+)
+
+// IntOrString holds a value that can be either a string or a int
+type IntOrString string
+
+// NullValue is an empty IntOrString value
+const NullValue = IntOrString("")
```

Comment:
Seems like there are a few places where we check for "zeroness" using:

```go
svcPort == intstr.NullValue || (svcPort.IsInt() && svcPort.Int32() == 0)
```

Do you think it makes sense to add an `IsZero()` method?

Reply:
Yeah, looks better.